tune model serving framework: add field details for model configs (#1976)
opensearch-project/documentation-website: _ml-commons-plugin/model-serving-framework.md

`name`| string | The name of the model. |
`version` | string | The version number of the model. Since OpenSearch does not enforce a specific version schema for models, you can choose any number or format that makes sense for your models. |
`model_format` | string | The portable format of the model file. Currently only supports `TORCH_SCRIPT`. |

> [`model_config`](#the-model_config-object) | string | The model's configuration, including the `model_type`, `embedding_dimension`, and `framework_type`. |

Reviewer: ```suggestion
[`model_config`](#the-model_config-object) | json object | The model's configuration, including the `model_type`, `embedding_dimension`, and `framework_type`. |
```

`model_config` | string | The model's configuration, including the `model_type`, `embedding_dimension`, and `framework_type`. |
`url` | string | The URL where the model is located. |

`model_config` fields

> | Field                 | Data Type         | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                         |

Reviewer 2: ```suggestion
| Field | Data Type | Description |
```

> | :-------------------- | :---------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

Reviewer 2: ```suggestion
| :--- | :--- | :--- |
```

> | `model_type`          | string            | model type like `bert`, If you are using Huggingface model, you can find model type in config.json, refer to this [Huggingface model](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2/blob/main/config.json#L15)                                                                                                                                                                                                                                      |

Reviewer 2: ```suggestion
| `model_type` | string | The model type (for example, `bert`). For a Huggingface model, the model type is specified in `config.json`. For an example, see the [`all-MiniLM-L6-v2` Huggingface model `config.json`](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2/blob/main/config.json#L15).|
```

| `embedding_dimension` | integer           | The dimension of model generated dense vector. If you are using Huggingface model, you can find on model card. For example this [Huggingface model](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) `384 dimensional dense vector space`, 384 is the embedding dimension.                                                                                                                                                                            |

> | `framework_type`      | string            | The framework model is using. For now, we support `sentence_transformers` or `huggingface_transformers`. For `sentence_transformers` model, it will output text embedding directly, so ml-commons won't do post processing. For `huggingface_transformers`, ml-commons will do post processing by applying mean pooling to get text embedding. Read this [sample huggingface model](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) for more details |

Reviewer 2: ```suggestion
| `framework_type` | string  | The framework the model is using. Currently, we support `sentence_transformers` and `huggingface_transformers` frameworks. The `sentence_transformers` model outputs text embeddings directly, so ML Commons does not perform any post processing. For `huggingface_transformers`, ML Commons performs post processing by applying mean pooling to get text embeddings. See the example [`all-MiniLM-L6-v2` Huggingface model](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) for more details. |
```

> | `all_config`          | string (optional) | This fied is just for reference purpose. You can put all model configuration to this field. For example, if you are using Huggingface model, you can minify the config.json file to one line and save all the content in `all_config` field. Once model uploaded, you can use get model API to get all model configurations stored in this field.                                                                                                                   |

Reviewer 2: ```suggestion
| `all_config` | string | Optional. This field is used for reference purposes. You can specify all model configurations in this field. For example, if you are using a Huggingface model, you can minify the `config.json` file to one line and save its contents in the `all_config` field. Once the model is uploaded, you can use the get model API operation to get all model configurations stored in this field. |
```
